src/qkit/gui/qviewkit/main.py

```diff
@@ -146,29 +146,15 @@ def main(argv=sys.argv):
 
 
     parser.add_argument('-f','--file',     type=str, help='hdf filename to open')
-    parser.add_argument('-u','--url', type=str, help='url handler for qviewkit:UUID scheme. Derives file location from fid.')
     parser.add_argument('-ds','--datasets', type=str, help='(optional) datasets opened by default')
 
     parser.add_argument('-rt','--refresh_time', type=float, help='(optional) refresh time')
     parser.add_argument('-sp','--save_plot',  default=False,action='store_true', help='(optional) save default plots')
     parser.add_argument('-live','--live_plot',default=False,action='store_true', help='(optional) if set, plots are reloaded')
     parser.add_argument('-qinfo','--qkit_info',default=False,action='store_true', help='(optional) if set, listen to qkit infos')
-    args=parser.parse_args()
+    args=parser.parse_args(args=argv[1:])
```

Comment:
This is the simplest change to allow for calling with non-argv arguments.
`parse_args()`, when called without arguments, behaves like this. But since this method allow overwriting of the parameter `argv`, we need to take the the data from there.

This single-line change allows for a URL-Handling wrapper, as implemented in [dokuwiki-autodoc](https://github.com/qkitgroup/dokuwiki-autodoc/blob/f0f26bc981e78ba6f25a715ae23e2b60426bb9f7/src/dokuwiki_autodoc/qviewkit_url_handler.py#L9)